### PR TITLE
Included Chl to soca state as a generic tracer

### LINKED ISSUE #349: Include Chl to the State as a generic tracer
## Description
First step towards biogeochem DA: Include a Chl state variable in soca, do not refer to BLING at this stage.

## Solution
- Update restart to include the necessary tracer(s)
- Add Chl to the soca **State** class 
- soca Model class: get Chl from the model. Note that this step is currently not necessary for EMC
- Add generic tracers to the current mom6 configuration

## Dependencies
None if not referring to BLING and not touching the Model class implementation (bullet 3 above).

### PR DESCRIPTION
**Pull request by liuxiao37k**
## Description

This PR included necessary changes in soca_fields_mod.F90, CMakelists.txt, and the soca test folder to include chlorophyll to soca state as generic tracer. At a first step, chlorophyll is only read from an external file (MOM.res.nc) and not retrieved from the simulated field in mom6/ocean_bgc.

Chlorophyll ('chl') is added to forecast_identity ctest.

## Issue(s) addressed
closes #349 

## Testing

All soca ctests passed on Hera.

## Dependencies
None

### CODE REVIEW

**Comment by liuxiao37k** [comment]
I closed the old PR (https://github.com/JCSDA/soca/pull/335) but kept the feature/ocean_bgc branch as it involves ongoing development to allow chlorophyll values to be passed between soca and mom6/BLING.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ list( APPEND soca_test_input
   testinput/forecast_identity.yml
   testinput/forecast_identity_cice.yml
   testinput/forecast_mom6.yml
+  testinput/forecast_mom6_bgc.yml
```
Comment (guillaumevernieres) [inline test/CMakeLists.txt]: This test isn't needed just yet. Move Chl to the **forecast_identity.yml** test

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +70,7 @@ list( APPEND soca_test_ref
   testref/forecast_identity_cice.test
   testref/forecast_identity.test
   testref/forecast_mom6.test
+  testref/forecast_mom6_bgc.test
```
Comment (guillaumevernieres) [inline test/CMakeLists.txt]: Same comment as above and adjust testref/forecast_identity.test

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -109,6 +111,8 @@ list( APPEND soca_obs
   Data/Obs/sst.nc
   Data/Obs/gmi_gpm_geoval.nc
   Data/Obs/gmi_gpm_obs.nc
+  Data/Obs/modis_aqua_oc_l2.nc
+  Data/Obs/viirs_jpss1_oc_l2.nc
```
Comment (guillaumevernieres) [inline test/CMakeLists.txt]: modis and viirs are not needed yet, remove.

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -347,6 +351,12 @@ soca_add_test( NAME forecast_mom6
                NOTRAPFPE
                TEST_DEPENDS test_soca_gridgen )
 
+# Test mom6_bgc forecast
+soca_add_test( NAME forecast_mom6_bgc
+               EXE  soca_forecast.x
+               NOTRAPFPE
+               TEST_DEPENDS test_soca_gridgen )
+
```
Comment (guillaumevernieres) [inline test/CMakeLists.txt]: same comment as above. Test not needed yet

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
@liuxiao37k : Some left over from your draft pr that needs to be cleaned up.

**Comment by guillaumevernieres** [comment]
@liuxiao37k , that work is ready to review/merge after you address a few minor issues, then you can move on to working on modis/viirs ufo.

**Comment by liuxiao37k** [comment]
@guillaumevernieres thanks for the comments. updated as asked. Once this is confirmed and merged, I will push a new branch with chl hofx that has been developed.

**Review by guillaumevernieres — APPROVED** [review]
Thanks @liuxiao37k

**Pull request merged**